build.sh

```diff
@@ -0,0 +1,140 @@
+   clean  - remove all existing build artifacts and configuration (start over). If clean is not used previous workspace(if available) will be used
+   librmm - build the librmm C++ code only
+   rmm    - build the rmm Python package
+   librmm_cffi - build and install the librmm_cffi package
+   -v     - verbose build mode
+   -g     - build for debug
+   -n     - no install step
+   -h     - print this text
+
+   default action (no args) is to build and install 'librmm' target
+"
+LIBRMM_BUILD_DIR=${REPODIR}/build
+RMM_BUILD_DIR=${REPODIR}/python/build
+BUILD_DIRS="${LIBRMM_BUILD_DIR} ${RMM_BUILD_DIR}"
+
+# Set defaults for vars modified by flags to this script
+VERBOSE=""
+BUILD_TYPE=Release
+INSTALL_TARGET=install
+
+# Set defaults for vars that may not have been defined externally
+#  FIXME: if INSTALL_PREFIX is not set, check PREFIX, then check
+#         CONDA_PREFIX, but there is no fallback from there!
+INSTALL_PREFIX=${INSTALL_PREFIX:=${PREFIX:=${CONDA_PREFIX}}}
+PARALLEL_LEVEL=${PARALLEL_LEVEL:=""}
+
+function hasArg {
+    (( NUMARGS != 0 )) && (echo " ${ARGS} " | grep -q " $1 ")
+}
+
+function createBuildArea {
+    if [ -d "${LIBRMM_BUILD_DIR}" ]; then
+        echo "Existing build directory detected. Skipping running cmake. To do clean build pass option clean"
+        cd "${LIBRMM_BUILD_DIR}"
+        return
+    fi
+    mkdir -p "${LIBRMM_BUILD_DIR}"
+    cd "${LIBRMM_BUILD_DIR}"
+    cmake -DCMAKE_INSTALL_PREFIX="${INSTALL_PREFIX}" \
+          -DCMAKE_CXX11_ABI=ON \
+          -DCMAKE_BUILD_TYPE=${BUILD_TYPE} ..
+}
+
+if hasArg -h; then
+    echo "${HELP}"
+    exit 0
+fi
+
+# Check for valid usage
+if (( NUMARGS != 0 )); then
+    for a in ${ARGS}; do
+	if ! (echo " ${VALIDARGS} " | grep -q " ${a} "); then
+	    echo "Invalid option: ${a}"
+	    exit 1
+	fi
+    done
+fi
+
+# Process flags
+if hasArg -v; then
+    VERBOSE=1
+    set -x
+fi
+if hasArg -g; then
+    BUILD_TYPE=Debug
+fi
+if hasArg -n; then
+    INSTALL_TARGET=""
+fi
+
+# If clean given, run it prior to any other steps
+if hasArg clean; then
+    # If the dirs to clean are mounted dirs in a container, the
+    # contents should be removed but the mounted dirs will remain.
+    # The find removes all contents but leaves the dirs, the rmdir
+    # attempts to remove the dirs but can fail safely.
+    for bd in ${BUILD_DIRS}; do
+        if [ -d "${bd}" ]; then
+            find "${bd}" -mindepth 1 -delete
+            rmdir "${bd}" || true
+	fi
+    done
+fi
+
+################################################################################
+# Configure, build, and install librmm
+if (( NUMARGS == 0 )) || hasArg librmm; then
+    echo "building librmm..."
+    createBuildArea
+    make -j${PARALLEL_LEVEL} VERBOSE=${VERBOSE}
+    if [[ ${INSTALL_TARGET} != "" ]]; then
+        make -j${PARALLEL_LEVEL} VERBOSE=${VERBOSE} install
+    fi
+fi
+
+# Build and install the rmm Python package
+if hasArg rmm; then
```

Comment:
I *think* the `cmake` step creates a file that these commands need. Can we combine the `librmm_cffi` target with this one, since they really don't make sense to run separately anyway?

Reply:
Fixed. Collapsed rmm and librmm_cffi into single target